juju/client/connection.py

```diff
@@ -26,7 +27,7 @@
     'AllWatcher': {'versions': [1, 2, 3]},
     'Annotations': {'versions': [2]},
     'Application': {'versions': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
-                                 14, 15]},
+                                 14, 15, 16]},
```

Comment:
This should be Application Facade `v17`, right?

Reply:
No. Finally, it's 16. See [here](https://github.com/juju/juju/blob/173c214fe69660ef9597cae44883d1ce330d7a62/apiserver/facades/client/application/application.go#L63)

Reply:
we should look to support 17 when it is released, but we can also just support the old method on 14,15,16 (note that we can eventually drop 14 as that is a 2.9 vs 3.0)